src/main/java/ch/njol/skript/sections/LoopSection.java

```diff
@@ -0,0 +1,164 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.sections;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.Section;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.lang.util.LyingItem;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+
+public class LoopSection extends Section implements LyingItem {
+
+	static {
+		Skript.registerSection("loop", LoopSection.class, "loop %objects%");
+	}
+
+	@SuppressWarnings("null")
+	private Expression<Object> loopedExpr;
+
+
+	@Nullable
+	private TriggerItem actualNext;
+
+	private Map<Event, Iterator<?>> iterators = new WeakHashMap<>();
+	private Map<Event, Object> values = new WeakHashMap<>();
+
+	@Nullable
+	private Iterator<?> getIterator(Event e) {
+		Iterator<?> iterator = iterators.get(e);
+		if (iterator == null) {
+			iterator = loopedExpr instanceof Variable<?> ?
+					((Variable<?>) loopedExpr).variablesIterator(e)
+					: loopedExpr.iterator(e);
```

Comment:
Please stop with this excessive line breaking. this only needs to be on a single line, this gets too ridiculous.

Reply:
I'll change it however what I'm actually trying to do is follow what was done in other PRs where nobody had an issue with the spacing